lib/compiler/aro_translate_c.zig

```diff
@@ -215,80 +218,292 @@ fn prepopulateGlobalNameTable(c: *Context) !void {
+                        if (field_size > max_field_alignment_bits)
+                            max_field_alignment_bits = field_size;
+                    }
+                }
+                if (max_field_alignment_bits != parent_ptr_alignment_bits) {
+                    break :head_align parent_ptr_alignment;
+                } else {
+                    break :head_align null;
+                }
+            } else {
+                break :head_align null;
+            }
+        };
+
+        // Iterate over field nodes so that we translate any type decls included in this record decl.
+        // TODO: Move this logic into `fn transType()` instead of handling decl translation here.
+        for (field_nodes) |field_node| {
+            const field_raw_ty = node_types[@intFromEnum(field_node)];
+            if (field_raw_ty.isEnumOrRecord()) try transDecl(c, scope, field_node);
+        }
+
+        for (record_decl.fields, 0..) |field, field_index| {
+            const field_loc = field.name_tok;
+
+            // Demote record to opaque if it contains a bitfield
+            if (!field.isRegularField()) {
+                try c.opaque_demotes.put(c.gpa, @intFromPtr(record_decl), {});
+                try warn(c, scope, field_loc, "{s} demoted to opaque type - has bitfield", .{container_kind_name});
+                break :blk ZigTag.opaque_literal.init();
+            }
+
+            var field_name = c.mapper.lookup(field.name);
+            if (!field.isNamed()) {
+                field_name = try std.fmt.allocPrint(c.arena, "unnamed_{d}", .{unnamed_field_count});
+                unnamed_field_count += 1;
+            }
+            const field_type = transType(c, scope, field.ty, .preserve_quals, field_loc) catch |err| switch (err) {
+                error.UnsupportedType => {
+                    try c.opaque_demotes.put(c.gpa, @intFromPtr(record_decl), {});
+                    try warn(c, scope, 0, "{s} demoted to opaque type - unable to translate type of field {s}", .{
+                        container_kind_name,
+                        field_name,
+                    });
+                    break :blk ZigTag.opaque_literal.init();
+                },
+                else => |e| return e,
+            };
+
+            const field_alignment = if (has_alignment_attributes)
+                alignmentForField(record_decl, head_field_alignment, field_index)
+            else
+                null;
+
+            // C99 introduced designated initializers for structs. Omitted fields are implicitly
+            // initialized to zero. Some C APIs are designed with this in mind. Defaulting to zero
+            // values for translated struct fields permits Zig code to comfortably use such an API.
+            const default_value = if (raw_record_ty.is(.@"struct"))
+                try ZigTag.std_mem_zeroes.create(c.arena, field_type)
+            else
+                null;
+
+            try fields.append(.{
```

Comment:
`fields` was created with `initCapacity` of the number of fields, so you should be able to do `appendAssumeCapacity` here.